src/services/EcoBridge/Socket/SocketBridge.ts

```diff
@@ -0,0 +1,584 @@
+        { signal: this.renewAbortController('quote') }
+      )
+    } catch (e) {
+      //if status of response isn't 20 (aborted) then set socket status to failed
+      const isValid = getStatusOfResponse(e)
+      if (!isValid) {
+        this.store.dispatch(
+          this.actions.setBridgeDetailsStatus({
+            status: SyncState.FAILED,
+            errorMessage: 'No available routes / details'
+          })
+        )
+        return
+      }
+    }
+
+    if (!quote) return
+
+    const { success, result } = quote
+
+    if (!success || result.routes.length === 0) {
+      this.store.dispatch(
+        this.actions.setBridgeDetailsStatus({
+          status: SyncState.FAILED,
+          errorMessage: 'No available routes / details'
+        })
+      )
+      return
+    }
+    const { routes, toAsset } = result
+
+    this.store.dispatch(this.actions.setRoutes(routes))
+
+    let tokenData: TokenPriceResponseDTO | undefined
+
+    try {
+      tokenData = await ServerAPI.appControllerGetTokenPrice({
+        tokenAddress: toAsset.address,
+        chainId: toAsset.chainId
+      })
+    } catch (e) {}
+
+    const bestRoute = getBestRoute(routes, tokenData, toAsset.decimals)
+
+    if (!bestRoute) {
+      this.store.dispatch(this.actions.setBridgeDetailsStatus({ status: SyncState.FAILED }))
+      return
+    }
+
+    const { toAmount, serviceTime, totalGasFeesInUsd, routeId } = bestRoute
+
+    this.store.dispatch(commonActions.setActiveRouteId(routeId))
+
+    const details = {
+      gas: `${totalGasFeesInUsd.toFixed(2).toString()}$`,
+      estimateTime: `${(serviceTime / 60).toFixed(0).toString()} min`,
+      receiveAmount: Number(formatUnits(toAmount, toAsset.decimals))
+        .toFixed(2)
+        .toString(),
+      requestId: helperRequestId
+    }
+
+    this.store.dispatch(this.actions.setBridgeDetails(details))
+  }
+
+  public fetchDynamicLists = async () => {
+    const {
+      from: { chainId: fromChainId },
+      to: { chainId: toChainId }
+    } = this.store.getState().ecoBridge.ui
+
+    if (!fromChainId || !toChainId) return
+
+    this.store.dispatch(this.actions.setTokenListsStatus(SyncState.LOADING))
+
+    const tokenListKey = `${Math.min(Number(fromChainId), Number(toChainId))}-${Math.max(
+      Number(fromChainId),
+      Number(toChainId)
+    )}`
+
+    const tokenList: TokenList = {
+      name: 'Socket',
+      timestamp: new Date().toISOString(),
+      version: {
+        major: 1,
+        minor: 0,
+        patch: 0
+      },
+      tokens: SOCKET_TOKENS[tokenListKey] ?? [],
+      logoURI: SocketLogo
+    }
+
+    this.store.dispatch(this.actions.addTokenLists({ socket: tokenList }))
+    this.store.dispatch(this.actions.setTokenListsStatus(SyncState.READY))
+  }
+
+  public fetchStaticLists = async () => {
+    this.store.dispatch(commonActions.activateLists(['socket']))
+  }
+
+  private startListeners = () => {
+    this._listeners.push(setInterval(this.pendingTxListener, 5000))
```

Comment:
talking about magic numbers, can't we have this 5000 read from a config file or env,? ... or even any other alternative

Reply:
Good catch. I think we will make a separate PR for all listeners within Swapr